File: pkg/judge/evidence.go
package judge

import (
	"github.com/smg247/prow-agent-eval/pkg/config"
	ghclient "github.com/smg247/prow-agent-eval/pkg/github"
)

// CaseEvidence is the collected evidence about what the agent did for one case.
// Judges and reporters consume it after Collect runs.
type CaseEvidence struct {
	GitHub      GitHubData
	Annotations config.CaseAnnotations
	BuildResult MakeResult
	TestResult  MakeResult
}

// MakeResult is the outcome of a make target run during collect.
type MakeResult struct {
	Collected bool // false = collect skipped / not run
	Passed    bool
	Output    string
	Error     string
}

// GitHubData holds PR, branch, and diff evidence gathered from GitHub/git.
type GitHubData struct {
	Repo                 string
	BaseBranch           string
	AgentBranch          string
	ExpectedBranch       string
	PRNumber             int
	PRState              string
	PRBody               string
	PRDescriptionFile    bool
	ChangedFiles         []string
	ExpectedChangedFiles []string
	FullDiff             string
	ExpectedFullDiff     string
	AgentDiffLines       int
	ExpectedDiffLines    int
	HasExpectedDiff      bool // set when expected_branch_diff collect ran
	BotReplies           []BotReply
	PostedComments       map[string]ghclient.PostedComment
}

// BotReply is a comment attributed to the configured bot login.
type BotReply struct {
	ID        int64
	Body      string
	CreatedAt string
	Path      string
	Type      string // "issue" | "review"
}

File: prow_agent_eval/evidence.py
"""Evidence dataclasses and serialization for judge consumption."""

from __future__ import annotations

from dataclasses import asdict, dataclass, field


@dataclass
class MakeResult:
    collected: bool = False
    passed: bool = False
    output: str = ""
    error: str = ""


@dataclass
class BotReply:
    id: int = 0
    body: str = ""
    created_at: str = ""
    path: str = ""
    type: str = ""


@dataclass
class GitHubData:
    repo: str = ""
    base_branch: str = ""
    expected_branch: str = ""
    agent_branch: str = ""
    pr_number: int = 0
    pr_body: str = ""
    changed_files: list[str] = field(default_factory=list)
    expected_changed_files: list[str] = field(default_factory=list)
    full_diff: str = ""
    expected_full_diff: str = ""
    bot_replies: list[BotReply] = field(default_factory=list)
    posted_comments: dict = field(default_factory=dict)


@dataclass
class CaseEvidence:
    github: GitHubData = field(default_factory=GitHubData)
    annotations: dict = field(default_factory=dict)
    build_result: MakeResult = field(default_factory=MakeResult)
    test_result: MakeResult = field(default_factory=MakeResult)


def evidence_to_dict(ev: CaseEvidence) -> dict:
    return asdict(ev)
